sphinx/builders/html/ogp.py

```diff
@@ -0,0 +1,159 @@
+"""
+    sphinx.builders.html.ogp
+    ~~~~~~~~~~~~~~~~~~~~~~~~
+
+    OGP tag plug-in for HTML builders.
+
+    :copyright: Copyright 2007-2020 by the Sphinx team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+import posixpath
+from typing import Any, Dict, Generator, List
+
+from docutils import nodes
+from docutils.nodes import Element
+
+from sphinx.application import Sphinx
+from sphinx.util.osutil import canon_path
+
+
+class OGDescriptionCollector(nodes.SparseNodeVisitor):
+    """A description collector module for OGP metadata."""
+
+    def __init__(self, document: nodes.document) -> None:
+        super().__init__(document)
+        self.descriptions = []  # type: List[str]
+        self.sections = 0       # type: int
+
+    def visit_compact_paragraph(self, node: Element) -> None:
+        if node.get('toctree'):
+            # ignore toctree
+            raise nodes.SkipChildren
+
+    def visit_section(self, node: Element) -> None:
+        self.sections += 1
+        if self.sections >= 3:
+            # stop processing when the 3rd section found
+            raise nodes.StopTraversal
+
+    def visit_paragraph(self, node: Element) -> None:
+        # collect paragraph as a part of descriptions
+        self.descriptions.append(node.astext())
+
+
+class OGImageCollector(nodes.SparseNodeVisitor):
+    """An image collector module for OGP metadata.
+
+    .. note:: This will be removed on dropping docutils-0.15 support.
+              Since 0.16, this class will be replaced by
+              ``Element.traverse()``.
+    """
+    def __init__(self, document: nodes.document) -> None:
+        super().__init__(document)
+        self.image = None  # type: nodes.image
+
+    def visit_image(self, node: nodes.image) -> None:
+        self.image = node
+        raise nodes.StopTraversal
+
+
+def get_ogpmeta_from_context(context: Dict) -> Dict:
+    """Pick OGP metadata up from the metadata of the document."""
+    htmlmeta = context.get('meta') or {}
+    return {k: v for k, v in htmlmeta.items() if k.startswith('og:')}
+
+
+def apply_html_ogp_meta(app: Sphinx, docname: str, ogpmeta: Dict) -> None:
+    """Apply :confval:`html_ogp_meta` to document specific OGP metadata."""
+    for key, value in app.config.html_ogp_meta.items():
+        if key == 'og:image':
+            # og:image is considered as a relative path from the document
+            value, _ = app.env.relfn2path(value, docname)
+
+        ogpmeta.setdefault(key, value)
+
+
+def extract_ogpmeta_from_content(app: Sphinx, docname: str, context: Dict,
+                                 ogpmeta: Dict, doctree: nodes.document) -> None:
+    """Extract OGP metadata from contents."""
+    # fill system-default OGP metadata
+    ogpmeta.setdefault('og:type', 'website')
+    ogpmeta.setdefault('og:title', context['title'])
```

Comment:
`context['title']` contains html tag if it belongs `:numbered:` toctree.
Code for `context['title']` is this: https://github.com/sphinx-doc/sphinx/blob/5850d6b8f9569171d73f8bc5abbc9d646f5d4a77/sphinx/builders/html/__init__.py#L552

Rendered title_node is this:
```
(Pdb) self.render_partial(title_node)['title']
'<span class="section-number">1. </span>Installation'
```
and sphinx generate incorrect html tags:
```
<meta property="og:title" content="<span class="section-number">1. </span>Installation" />
```

(my original code also have this problem.. :(

Reply:
Okay, I'll take the title from `env.titles`.